cmd/helm/dependency.go

```diff
@@ -71,6 +71,36 @@ the dependency charts stored locally. The path should start with a prefix of
 If the dependency chart is retrieved locally, it is not required to have the
 repository added to helm by "helm add repo". Version matching is also supported
 for this case.
+
+A repository can be defined as a git URL. The path must start with a prefix of
+"git:" followed by a valid git repository URL.
+
+    # Chart.yaml
+    dependencies:
+    - name: helm-chart
+      version: "main"
```

Comment:
In general, I'm not sure how this will work as a version. In the OCI support, we are bound to use SemVer2 for now).

Can we instead rely on fetching a git tag that is properly semantically versioned? For example `0.1.0`